pkg/commands/componentarchive/signature/signature_test.go

```diff
@@ -0,0 +1,443 @@
+			parentCd := getParentCd()
+			childCd := getChildCd()
+
+			ref := fmt.Sprintf("%s/%s", testenv.Addr, "cd/0")
+			uploadTestCd(parentCd, ref)
+			uploadTestCd(childCd, ref)
+
+			//add digests
+			digestedCds, err := signatures.RecursivelyAddDigestsToCd(&parentCd, *cdv2.NewOCIRegistryRepository(ref, ""), client, map[string]ctf.BlobResolver{}, context.TODO(), map[string]bool{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(digestedCds)).To(Equal(2))
+
+			digestedParentCd := digestedCds[1]
+
+			//change access imageReference of child to wrong resource (=resource of parent) --> content=digest is different
+			refWrongResource := fmt.Sprintf("%s/%s", testenv.Addr, "cd/0/test-resource-parent:v0.0.1")
+			wrongResourceAccess, err := cdv2.NewUnstructured(cdv2.NewOCIRegistryAccess(refWrongResource))
+			Expect(err).ToNot(HaveOccurred())
+			childCd.Resources[0].Access = &wrongResourceAccess
+			uploadTestCd(childCd, ref)
+
+			repoCtx := cdv2.NewOCIRegistryRepository(ref, "")
+			err = verify.CheckCdDigests(digestedParentCd, *repoCtx, client, context.TODO())
+			Expect(err).To(HaveOccurred())
+		})
+		It("should fail verify with manipulated resource in parent", func() {
+			parentCd := getParentCd()
+			childCd := getChildCd()
+
+			ref := fmt.Sprintf("%s/%s", testenv.Addr, "cd/0")
+			uploadTestCd(parentCd, ref)
+			uploadTestCd(childCd, ref)
+
+			//add digests
+			digestedCds, err := signatures.RecursivelyAddDigestsToCd(&parentCd, *cdv2.NewOCIRegistryRepository(ref, ""), client, map[string]ctf.BlobResolver{}, context.TODO(), map[string]bool{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(digestedCds)).To(Equal(2))
+
+			digestedParentCd := digestedCds[1]
+
+			//change access imageReference of parent to wrong resource (= resource of child) --> content=digest is different
+			refWrongResource := fmt.Sprintf("%s/%s", testenv.Addr, "cd/0/test-resource-child:v0.0.1")
+			wrongResourceAccess, err := cdv2.NewUnstructured(cdv2.NewOCIRegistryAccess(refWrongResource))
+			Expect(err).ToNot(HaveOccurred())
+			digestedParentCd.Resources[0].Access = &wrongResourceAccess
+
+			repoCtx := cdv2.NewOCIRegistryRepository(ref, "")
+			err = verify.CheckCdDigests(digestedParentCd, *repoCtx, client, context.TODO())
+			Expect(err).To(HaveOccurred())
+		})
+		It("should fail verify with component reference digest manipulation", func() {
+			parentCd := getParentCd()
+			childCd := getChildCd()
+
+			ref := fmt.Sprintf("%s/%s", testenv.Addr, "cd/0")
+			uploadTestCd(parentCd, ref)
+			uploadTestCd(childCd, ref)
+
+			//add digests
+			digestedCds, err := signatures.RecursivelyAddDigestsToCd(&parentCd, *cdv2.NewOCIRegistryRepository(ref, ""), client, map[string]ctf.BlobResolver{}, context.TODO(), map[string]bool{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(digestedCds)).To(Equal(2))
+
+			digestedParentCd := digestedCds[1]
+
+			//manipulate digest of component descriptor
+			digestedParentCd.ComponentReferences[0].Digest.Value = "faked"
```

Comment:
maybe use some real digest to make the example more realistic?

Reply:
I dont see how it matters if it is a realistic digest or not. They only have to be different in at least one character to count as different.